contrib/python/systemtest/smoketest_version.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+
+"""
+Simple stand-alone system smoke-test against the podman version sub-command
+"""
+
+import unittest
+import subprocess
+
+
+class TestVersion(unittest.TestCase):
+
+    TIMEOUT_SECONDS = 5
+
+    @staticmethod
+    def fmt_popen(popen, stdout, stderr):
+        """
+        Format the exit code, stdout and stderr as a msg string for display
+        """
+        msg = '\n\tCMD: {0} \n\tEXIT: {1}\n\tSTDOUT: {2}\n\tSTDERR: {3}\n'
+        return msg.format(popen.args, popen.returncode, stdout, stderr)
+
+    def test_extra_args(self):
+        """
+        Verify the 'podman version' subcommand ignores extranious positional arguments
+        """
+        cmd = 'sudo podman version with extra args - --'
+        was_killed = False
+        popen = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                                 stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                 close_fds=True)
+
+        try:
+            (stdout, stderr) = popen.communicate(timeout=self.TIMEOUT_SECONDS)
+        except subprocess.TimeoutExpired:
+            was_killed = True
+            popen.kill()
```

Comment:
You may need to do something like the following:

pro = subprocess.Popen(cmd, stdout=subprocess.PIPE,  shell=True, preexec_fn=os.setsid) 
os.killpg(os.getpgid(pro.pid), signal.SIGTERM)

The issue is shell=True and you wanting to kill off the shell AND podman.  If you use shell=False this is not an issue.

Reply:
Good catch, but a runny-nose has my understanding operating at 50% :face_with_head_bandage:  

I thought ``popen.kill()`` would kill the shell-process, that would then signal it's child-process (the podman command).  I will RTFM before fixing this so I understand what's happening.

Reply:
Ohhhhh, okay, wow.  That's really good to know.  Thanks @jwhonce